### PR TITLE
Add agency and request detail views

### PR DESCRIPTION
**Pull request by sarayourfriend**
Adds some preliminary views for the agency and record requests for us to work off of.

Captura de Tela 2021-07-03 às 06 06 20

Captura de Tela 2021-07-03 às 06 20 03

Captura de Tela 2021-07-03 às 06 28 37

### CODE REVIEW

File: pra_request_tracker/record_requests/models.py [hunk 1]
```diff
@@ -57,6 +59,14 @@ def delete(self):
         """
         pass
 
+    @property
+    def status_label(self):
+        return self.Status(self.status).label
+
+    @cached_property
+    def files(self):
+        return self.recordrequestfile_set.all()
```
Comment (AetherUnbound) [inline pra_request_tracker/record_requests/models.py:77]: :star:

**Comment by sarayourfriend** [comment]
Hrmmm... according to Django Debugging Toolbar the prefetches aren't working.

**Comment by sarayourfriend** [comment]
Actually I guess they _are_ working, just not how I want it to. `prefetch_related` will do separate queries and then do the joining in Python 🤔 In any case it ends up being only two queries instead of n queries so that's good.

Captura de Tela 2021-07-03 às 06 38 37

Results in the following queries:

Captura de Tela 2021-07-03 às 06 38 33

So we're good.

File: pra_request_tracker/record_requests/views.py [hunk 2]
```diff
@@ -1,3 +1,21 @@
-from django.shortcuts import render  # noqa
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.views.generic import DetailView
+from django.views.generic import ListView
 
-# Create your views here.
+from .models import Agency, RecordRequest
+
+
+class AgencyListView(LoginRequiredMixin, ListView):
```
Comment (AetherUnbound) [inline pra_request_tracker/record_requests/views.py]: Do we want login required for these? I think they might be useful to the public

File: pra_request_tracker/templates/base.html [hunk 3]
```diff
@@ -68,6 +68,9 @@
             <li class="nav-item">
               <a class="nav-link" href="{% url 'about' %}">About</a>
             </li>
+            <li class="nav-item">
+              <a class="nav-link" href="{% url 'record_requests:agencies' %}">Agencies</a>
```
Comment (AetherUnbound) [inline pra_request_tracker/templates/base.html:72]: TIL about URL namespaces!

File: pra_request_tracker/templates/record_requests/agency_detail.html [hunk 4]
```diff
@@ -0,0 +1,44 @@
+{% extends "base.html" %}
+{% load static %}
+
+{% block title %}Agency: {{object.name}}{% endblock %}
```
Comment (AetherUnbound) [inline pra_request_tracker/templates/record_requests/agency_detail.html]: Nit:
```suggestion
{% block title %}Agency: {{ object.name }}{% endblock %}
```

File: pra_request_tracker/templates/record_requests/agency_list.html [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+          </tr>
+        </thead>
+        <tbody>
+          {% for agency in object_list %}
+            <tr>
+              <td>
+                <a href="{% url 'record_requests:agency-detail' name=agency.name %}">{{ agency.name }}</a>
+              </td>
+              <td>
+                {{ agency.request_count }}
```
Comment (AetherUnbound) [inline pra_request_tracker/templates/record_requests/agency_list.html:31]: Really great idea!!

**Review by AetherUnbound — APPROVED** [review]
Looking amazing! Just a few comments, but great otherwise

File: pra_request_tracker/record_requests/views.py (hunk 2 above)
Comment (sarayourfriend) [reply pra_request_tracker/record_requests/views.py]: True! I'll remove them, but I think we should require logging in for file downloads, what do you think?

Comment (AetherUnbound) [reply pra_request_tracker/record_requests/views.py]: Even then I'm not sure...I'd rather just make it easier for folks to get this data :thinking:

Comment (sarayourfriend) [reply pra_request_tracker/record_requests/views.py]: Okay! We'll just make it all open :D

**Pull request merged**